src/components/hook.js

```diff
@@ -26,10 +26,15 @@ export function createHook(Store, { selector } = {}) {
       [hasPropsArg, storeState]
     );
 
+    const forceUpdate = useState({})[1];
     const getSnapshot = useCallback(() => {
-      const state = retrieveStore(Store).storeState.getState();
+      // parent scope has changed and notify was explicitly triggered by the container
+      // we need to force the hook to re-render to listen new storeState
+      if (retrieveStore(Store).storeState !== storeState) forceUpdate({});
```

Comment:
This forces a re-render on scope change. Currently it might be "skipped" if `stateSelector` return value is === (possible due to aggressive memo)